File: js/button_error.js
// Rookie mistake using var: alerts 10 for every button.
for (var i = 0; i < 10; i++) {
  const button = document.createElement("button");
  button.innerHTML = i;
  button.addEventListener('click', e => {alert(i);});
  document.body.appendChild(button);
}

File: dart/button_error.dart
import 'dart:html';
void main() {
  for (var i = 0; i < 10; i++) {
    var button = new ButtonElement();
    button.text = i;
    button.onClick.listen((e) => window.alert(i));
    document.body.children.add(button);
  }
}
